packages/cli/src/backend-identifier/backend_identifier_resolver.ts

```diff
@@ -11,6 +13,9 @@ export type BackendIdentifierResolver = {
   resolve: (
     args: BackendIdentifierParameters
   ) => Promise<DeployedBackendIdentifier | undefined>;
+  resolveDeployedBackendIdToBackendId: (
+    deployedBackendId?: DeployedBackendIdentifier
+  ) => Promise<BackendIdentifier | undefined>;
```

Comment:
```suggestion
  resolveDeployedBackendIdToBackendId: (
    deployedBackendId: DeployedBackendIdentifier
  ) => Promise<BackendIdentifier>;
```

Should we have caller handle `undefined` ?

Or perhaps should we have 
```
resolveBackendIdentifier: (
    args: BackendIdentifierParameters
  ) => Promise<BackendIdentifier | undefined>;
```
as additional API ?

The thing is that if callers give this type `BackendIdentifierParameters` they should be getting final product instead of passing return value from one API to the other one - that solution is not much beneficial over just calling "Conversions". 

I like the later more as I type this.

Reply:
I'm all for the latter, it just makes me question the use of the original `resolve` for other commands but looks like those are different and do require `DeployedBackendIdentifier`.

Reply:
> for other commands but looks like those are different and do require DeployedBackendIdentifier.

Yep. that's why that api has to stay. Consider renaming it though to make it less confusing.